src/test/java/org/jpeek/metrics/Lcom4Test.java

```diff
@@ -0,0 +1,165 @@
+/**
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2017-2019 Yegor Bugayenko
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package org.jpeek.metrics;
+
+import org.cactoos.list.ListOf;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Test;
+
+/**
+ * Test case for LCOM4
+ * LCOM4 = Logical Relatedness of Methods.
+ * Basically it's a graph disjoint set.
+ * I.e. it answers the the following question:
+ * "Into how many independent classes can you split this class?"
+ * @author Ilya Kharlamov (ilya.kharlamov@gmail.com)
+ * @version $Id$
+ * @since 0.28
+ * @checkstyle JavadocMethodCheck (500 lines)
+ * @checkstyle ClassDataAbstractionCouplingCheck (500 lines)
+ */
+public final class Lcom4Test {
+
+    /**
+     * Variable from the report.
+     */
+    public static final String PAIRS = "pairs";
+
+    /**
+     * Variable from the report.
+     */
+    public static final String ATTRIBUTES = "attributes";
+
+    /**
+     * Variable from the report.
+     */
+    public static final String METHODS = "methods";
+
+    /**
+     * Test helper.
+     */
+    private MetricBase base;
```

Comment:
@ilyakharlamov The test methods should be independents and no share objects. Please, create a new `MetricBase` for each test.

Reply:
done